File: Native/Test/Fabric/event-handler.js
F = require('Fabric').createClient();

eh = F.DG.createEventHandler("event handler");
console.log(eh.getName());
console.log(eh.getType());

var mapNamedObjectsToNames = function (namedObjects) {
  var result = [];
  for (var i=0; i<namedObjects.length; ++i)
    result.push(namedObjects[i].getName());
  return result;
};


console.log(JSON.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))));
var childEventHandler = F.DG.createEventHandler("child event handler");
eh.appendChildEventHandler(childEventHandler);
console.log(JSON.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))));
eh.appendChildEventHandler(F.DG.createEventHandler("another child event handler"));
console.log(JSON.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))));
eh.removeChildEventHandler(childEventHandler);
console.log(JSON.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))));

F.flush();

File: Native/Test/Python/event-handler.py
import fabric
F = fabric.createClient()

eh = F.DG.createEventHandler("event handler")
print(eh.getName())
print(eh.getType())

def mapNamedObjectsToNames( objects ):
  result = []
  for i in range( 0, len( objects ) ):
    result.append( objects[i].getName() )
  return result

print(fabric.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))))

childEventHandler = F.DG.createEventHandler("child event handler")
eh.appendChildEventHandler(childEventHandler)
print(fabric.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))))

eh.appendChildEventHandler(F.DG.createEventHandler("another child event handler"))
print(fabric.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))))

eh.removeChildEventHandler(childEventHandler)
print(fabric.stringify((mapNamedObjectsToNames(eh.getChildEventHandlers()))))

F.close()
